File: android/walletsdk/src/main/java/leaf/prod/walletsdk/model/order/Ring.java
package leaf.prod.walletsdk.model.order;

import java.util.List;

import lombok.Data;

/**
 * Created with IntelliJ IDEA.
 * User: laiyanyan
 * Time: 2019-03-18 5:08 PM
 * Cooperation: loopring.org 路印协议基金会
 */
@Data
public class Ring {

    private String ringHash;

    private Long ringIndex;

    private Long fillsAmount;

    private String miner;

    private String txHash;

    private List<Fee> fees;

    private Long blockHeight;

    private Long blockTimestamp;

    public Ring convert() {
        if(fees != null) {
            for(Fee fee : fees) {
                fee.convert();
            }
        }
        return this;
    }
}

File: ios/loopr-ios/Models/Ring.swift
//
//  Ring.swift
//  loopr-ios
//
//  Created by ruby on 4/12/19.
//  Copyright © 2019 Loopring. All rights reserved.
//

import Foundation

class Ring {
    
    var ringHash: String = ""
    
    var ringIndex = 0

    var fillsAmount = 0
    
    var miner: String = ""
    
    var txHash: String = ""
    
    var fees: [FeeParams] = []
    
    var blockHeight = 0
    
    var blockTimestamp: Int = 0

    init(json: JSON) {
        ringHash = json["ringHash"].stringValue
        ringIndex = json["ringIndex"].intValue
        fillsAmount = json["fillsAmount"].intValue
        miner = json["miner"].stringValue
        txHash = json["txHash"].stringValue
        fees = []
        let arrayData = json["fees"].arrayValue
        for subJson in arrayData {
            let fee = FeeParams(json: subJson)
            fees.append(fee)
        }
        blockHeight = json["blockHeight"].intValue
        blockTimestamp = json["blockTimestamp"].intValue
    }

}
